org.eclipse.tm4e.ui/src/main/java/org/eclipse/tm4e/ui/text/TMPresentationReconciler.java

```diff
@@ -408,7 +408,7 @@ private IGrammar findGrammar(ContentTypeInfo info) {
 				// Discover the well grammar from the filetype
 				String fileName = info.getFileName();
 				if (fileName != null) {
-					String fileType = new Path(fileName).getFileExtension();
+					String fileType = "." /* #202 */ + new Path(fileName).getFileExtension();

```

Comment:
I don't think this line should be changed. Implementation of `getGrammarForFileType` should be fixed to receive `sh` or `.sh` and return the same result.
Maybe it would even be better to have the IGrammarRegistryManager have a getGrammarForFileName(...) instead that would take care of the logic of extracting the right subset of the file name.

Reply:
I quite agree on this, since I believe that some tmLanguage file define sometimes ".sh", sometimes "sh".